### PR TITLE
(Un)escape URLs when reading/writing `<img src=>` attributes

### LINKED ISSUE #2705: `build_article()` gets confused with percent-encoding in `<img>` src path
Articles cannot get built when when image files contain e.g. spaces in their absolute path names.

I think the percent-encoded path components are not correctly decoded before getting passed to `fs::path_real()`. Side note: `fs::path_real()` does not work on non-existing filesystem entries (see <https://github.com/r-lib/fs/issues/389>).

## Example

```r
reprex <- function(pattern) {
  d <- withr::local_tempdir(pattern = pattern)
  withr::local_dir(new = d)

  git2r::clone(url = "https://github.com/r-lib/pkgdown.git",
               local_path = ".",
               progress = FALSE)

  git2r::checkout(branch = "v2.1.0")

  devtools::test(filter = "^build-article$")
}

reprex(pattern = "pkgdown   reprex")
```

## Result from above
<pre>
Error (test-build-article.R:250:3): image links relative to output
<ENOENT/fs_error/error/condition>
Error: [ENOENT] Failed to realize '/tmp/RtmpaXzN7s/pkgdown%20%20%20reprex42618486606/tests/testthat/assets/articles-images/vignettes/another-kitten.jpg': no such file or directory
</pre>

### PR DESCRIPTION
**Pull request by HenningLorenzen-ext-bayer**
Fixes #2705 

- ~fix building on R 4.0.0~ (not applicable with {xml2} functions)
- [x] replace `utils::URLdecode()` and `utils::URLencode()`

### CODE REVIEW

File: R/tweak-page.R [hunk 1]
```diff
@@ -47,10 +47,16 @@ tweak_rmarkdown_html <- function(html, input_path, pkg = list(bs_version = 3)) {
   src <- xml2::xml_attr(img, "src")
   abs_src <- is_absolute_path(src)
   if (any(abs_src)) {
+    img_target_nodes <- img[abs_src]
+    img_src_real <- path_real(URLdecode(src[abs_src]))
```
Comment (hadley) [inline R/tweak-page.R:51]: Thanks for tackling this, but would you mind looking around for an alternative encoding/decoding function? (I feel like there has to be one in pkgdown's existing dependencies). The base functions are cripplingly slow. 

(Also no need to import these functions; just use with `::`.

Comment (HenningLorenzen-ext-bayer) [reply R/tweak-page.R]: Using now `{xml2}` functions.

File: R/check-built.R [hunk 2]
```diff
@@ -14,7 +14,7 @@ check_missing_images <- function(pkg, src_path, dst_path) {
   img <- xml2::xml_find_all(html, ".//img")
   src <- xml2::xml_attr(img, "src")
 
-  rel_src <- src[xml2::url_parse(src)$scheme == ""]
+  rel_src <- xml2::url_unescape(src[xml2::url_parse(src)$scheme == ""])
```
Comment (hadley) [inline R/check-built.R:17]: Perfect!

File: R/pkgdown-package.R [hunk 3]
```diff
@@ -2,9 +2,9 @@
 "_PACKAGE"
 
 ## usethis namespace: start
-#' @importFrom utils installed.packages
-#' @import rlang
 #' @import fs
+#' @import rlang
```
Comment (hadley) [inline R/pkgdown-package.R:6]: Would you mind reverting all the changes to this file?

File: tests/testthat/test-build-article.R [hunk 4]
```diff
@@ -276,6 +276,13 @@ test_that("warns about missing images", {
   expect_snapshot(build_article("kitten", pkg))
 })
 
+test_that("spaces in base paths do work", {
+  d <- withr::local_tempdir("basename with spaces")
+  fs::dir_copy(test_path("assets/articles-images"), d, overwrite = TRUE)
```
Comment (hadley) [inline tests/testthat/test-build-article.R:281]: Could you please use `local_pkgdown_site()` and `pkg_add_file()` as in the next test down?

**Review by hadley — COMMENTED** [review]
Thanks so much for working on this!

File: R/pkgdown-package.R (hunk 3 above)
Comment (HenningLorenzen-ext-bayer) [reply R/pkgdown-package.R]: Done in b4e10125

File: tests/testthat/test-build-article.R (hunk 4 above)
Comment (HenningLorenzen-ext-bayer) [reply tests/testthat/test-build-article.R]: Done: cc095ae2

File: tests/testthat/_snaps/build-article.new.md
Comment (HenningLorenzen-ext-bayer) [inline tests/testthat/_snaps/build-article.new.md:1]: I think `tests/testthat/_snaps/build-article.new.md` should be removed.

Comment (hadley) [reply tests/testthat/_snaps/build-article.new.md:1]: Oops, thanks!

**Comment by hadley** [comment]
@HenningLorenzen-ext-bayer thanks!

**Pull request merged**